device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionHandler.java

```diff
@@ -338,110 +338,112 @@ void openLinks()
+        //Check if the message being sent is a subscription change message. If so, open the corresponding links.
+        if (message instanceof IotHubTransportMessage)
+        {
+            IotHubTransportMessage transportMessage = (IotHubTransportMessage) message;
+            DeviceOperations subscriptionType = ((IotHubTransportMessage) message).getDeviceOperationType();
+
+            if (subscriptionType == DEVICE_OPERATION_METHOD_SUBSCRIBE_REQUEST)
+            {
+                if (this.methodsSenderLinkOpened && this.methodsReceiverLinkOpened)
                 {
-                    if (this.methodsSenderLinkOpened && this.methodsReceiverLinkOpened)
-                    {
-                        // No need to do anything. Method links are already opened
-                        this.amqpsSessionStateCallback.onMessageAcknowledged(message, Accepted.getInstance(), this.getDeviceId());
-                        return true;
-                    }
+                    // No need to do anything. Method links are already opened
+                    this.amqpsSessionStateCallback.onMessageAcknowledged(message, Accepted.getInstance(), this.getDeviceId());
+                    return SendResult.SUCCESS;
+                }
 
-                    if (this.explicitInProgressMethodsSubscriptionMessage == null)
-                    {
-                        createMethodLinks();
-                        this.explicitInProgressMethodsSubscriptionMessage = transportMessage;
-                        return true; //connection layer doesn't care about this delivery tag
-                    }
-                    else
-                    {
-                        log.debug("Rejecting methods subscription message because that subscription is already in progress");
-                        return false;
-                    }
+                if (this.explicitInProgressMethodsSubscriptionMessage == null)
+                {
+                    createMethodLinks();
+                    this.explicitInProgressMethodsSubscriptionMessage = transportMessage;
+                    return SendResult.SUCCESS; //connection layer doesn't care about this delivery tag
                 }
-                else //noinspection StatementWithEmptyBody
-                    if (subscriptionType == DEVICE_OPERATION_TWIN_UNSUBSCRIBE_DESIRED_PROPERTIES_REQUEST)
+                else
                 {
-                    //TODO: can add logic here to tear down twin links if the user wants to unsubscribe to desired properties
+                    log.debug("Rejecting methods subscription message because that subscription is already in progress");
+                    return SendResult.DUPLICATE_SUBSCRIPTION_MESSAGE;
                 }
-                else if (subscriptionType == DEVICE_OPERATION_TWIN_SUBSCRIBE_DESIRED_PROPERTIES_REQUEST)
+            }
+            else //noinspection StatementWithEmptyBody
```

Comment:
I don't know what to make of this comment

Reply:
This is a warnings suppression, but I can remove it if I comment out this else if block